gulpfile.ts

```diff
@@ -0,0 +1,38 @@
+import del = require("del");
+import * as gulp from "gulp";
+const httpServer = require("http-server");
+import * as path from "path";
+import ts = require("gulp-typescript");
+
+const out = "public";
+function outDir(name: string): string { return path.join(out, name); }
+
+gulp.task("clean", () => del(`${out}/*`));
+
+gulp.task("script", ["clean"], () => {
+    const tsProject = ts.createProject("assets/script/tsconfig.json", {
+        typescript: require("typescript")
+    });
+    const tsResult = tsProject.src().pipe(ts(tsProject));
```

Comment:
Consider `return tsProject.src().pipe(ts(tsProject)).js.pipe(gulp.dest(outDir("script")))` if you're not actually reusing `tsResult`.